src/Four_Player_Adapter.md

```diff
@@ -81,117 +175,151 @@ software point of view. Because of the way the DMG-07 hardcodes player IDs
+5           | P3.1 (byte 1)             | 0             | 0             | 0             | 0 
+6           | P3.1 (byte 2)             | 0             | 0             | 0             | 0
+7           | P4.1 (byte 1)             | 0             | 0             | 0             | 0 
+8           | P4.1 (byte 2)             | 0             | 0             | 0             | 0
+
+
+All connected Game Boys should send their data into the buffer during the first few serial
+transfers. Here, the packet size is 2 bytes, so each Game Boy should submit their data
+during the first 2 transfers. The other 6 transfers don't care what the Game Boys send;
+it won't enter into the buffer. The next 8 transfers return the data each Game Boy
+previously sent (if no Game Boy exists for a player, that slot is filled with zeroes).
 
 When the DMG-07 enters the transmission phase, the buffer is initially filled
 with garbage data that is based on output the master Game Boy had sent during
-the ping phase. At this time, it is recommended to ignore the earliest packets
+the ping phase. At this time, it is recommended to ignore the first packet
 received, however, it is safe to start putting new, relevant data into the
-buffer.
+buffer immediately.
 
-### Restarting Ping Phase
+## Restarting Ping Phase
 
 It's possible to restart the ping phase while operating in the transmission
-phase. To do so, the master Game Boy should send 4 or more bytes
-(`FF FF FF FF`, it's possible fewer \$FF bytes need to be sent,
-but this has not been extensively investigated yet). The bytes alert the DMG-07
-that the ping phase should begin again, after which it sends ping packets after
-a brief delay. During this delay, the transmission protocol is still working as
-intended until the switch happens.
+phase. To do so, any connected Game Boy can send the ping restart sequence,
+which is 3 or more \$FF bytes in a row (`FF FF FF FF`). The transmission
+should be aligned such that the first byte is sent in reply to the first
+serial transfer of a packet. This causes the DMG-07 to switch back to the
+ping phase.
+
+After the third consecutive \$FF byte is received by the DMG-07 and transmission
+of the current packet is completed then the DMG-07 will begin transmitting a
+packet where all the bytes are set to \$FF. This is the indicator that connected
+Game Boys should use for when to switch back to ping phase.
+
+To avoid false positives the Game Boys should only perform a switch to ping after
+receving an entire packet of consecutive \$FF bytes. For example, if the SIZE
+setting is 3 then the total packet size is 12 bytes (3×4), which will be the
+number of consecutive \$FF bytes to require from the DMG-07 to indicate a switch.
+
+After the DMG-07 finishes sending the indicator packet of \$FF bytes it will
+immediately begin transmitting ping phase packets.
+
+The following chart is an example of switching from transmission back to ping phase.
```

Comment:
It feels like an introduction to the list is missing here?

The following chart [...] then a list is presented